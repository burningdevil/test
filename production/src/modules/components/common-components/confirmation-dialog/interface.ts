export type ConfirmationDialogWordings = {
  title: string
  actionButtonText: string
  cancelButtonText: string
  summaryText: string
  detailText?: string
}

export interface ConfirmationDialogProps {
  isConfirmationDialogDisplayed: boolean
  triggerAction: () => void
  closeDialog: () => void
  wordings: ConfirmationDialogWordings
}
