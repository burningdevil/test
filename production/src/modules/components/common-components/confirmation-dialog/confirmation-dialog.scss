@import '~@mstr/rc/es/assets/scss/variables.scss';

$prefix: confirmation-dialog;

.#{$prefix} {
  width: 500px !important;

  .mstr-rc-dialog-title {
    @include msdl-h3;
    background-color: $msdl-white;
  }

  border: 1px solid $msdl-border-gray;
  box-shadow: 0 1px 4px 0 $msdl-shadow-gray;
  border-radius: $msdl-border-radius;

  .#{$prefix}-dialog-container {
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    background-color: $msdl-fill-gray;

    .#{$prefix}-warning-icon {
      padding: 25px;
    }

    .#{$prefix}-content-container {
      padding: 30px 15px 30px 0px;
      max-height: 320px;
      display: flex;
      flex-direction: column;
      align-items: flex-start;
      @include msdl-paragraph;

      .#{$prefix}-summary {
        padding-bottom: 5px;
        font-weight: bold;
      }

      .#{$prefix}-details {
        overflow: scroll;


        &::-webkit-scrollbar {
          display: none;
        }
      }
    }
  }

  .mstr-rc-dialog-footer {
    padding-bottom: 15px;
    padding-right: 10px;
    background-color: $msdl-fill-gray;
  }
}
