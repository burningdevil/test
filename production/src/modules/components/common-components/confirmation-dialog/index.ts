// eslint-disable-next-line import/prefer-default-export
export { ConfirmationDialog } from './confirmation-dialog'
// eslint-disable-next-line
// export {ConfirmationDialogWordings, ConfirmationDialogProps} from './interface'
