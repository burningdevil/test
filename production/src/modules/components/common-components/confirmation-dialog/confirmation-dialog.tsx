import * as React from 'react'

/* HOC */
import { InformationDialog } from '@mstr/rc'

/* Type */
import { ButtonConfigTypeDef } from '@mstr/rc/types/information-dialog/interface'
import { ConfirmationDialogProps } from './interface'

import './confirmation-dialog.scss'

export const ConfirmationDialog = (props: ConfirmationDialogProps) => {
  const { isConfirmationDialogDisplayed, triggerAction, closeDialog, wordings } = props

  /* State */

  /* Event Handler */
  const handleAction = () => {
    triggerAction()
    closeDialog()
  }

  const handleCancel = () => {
    closeDialog()
  }

  const handleWindowClose = () => {
    handleCancel()
  }

  /* ClassName */
  const prefix = 'confirmation-dialog'

  const buttonConfig = [
    {
      title: wordings.actionButtonText,
      className: `${prefix}-action-button`,
      onClick: handleAction,
      type: 'primary',
      size: 'regular',
    },
    {
      title: wordings.cancelButtonText,
      className: `${prefix}-cancel-button`,
      onClick: handleCancel,
    },
  ]

  /* Sub-Components */

  const content = (
    <div className={`${prefix}-dialog-container`}>
      <div className={`${prefix}-warning-icon`}>
        <i className="ws-icon-60 ws-icon-mstr-warning" />
      </div>
      <div className={`${prefix}-content-container`}>
        <div tabIndex={0} className={`${prefix}-summary`}>
          <span>{wordings.summaryText}</span>
        </div>
        <div tabIndex={0} className={`${prefix}-details`}>
        <span>{wordings.detailText}</span>
        </div>
      </div>
    </div>
  )

  return (
      <div className={`${prefix}-wrapper`}>
        <InformationDialog
          isOpen={isConfirmationDialogDisplayed}
          title={wordings.title}
          hideTitle={false}
          content={content}
          buttonConfig={buttonConfig as ButtonConfigTypeDef[]}
          appElement={document.querySelector(`.${prefix}-wrapper`)}
          className={prefix}
          ariaLabel={wordings.title}
          theme= ""
          onClose={handleWindowClose}
        />
      </div>
  )
}

export default ConfirmationDialog
