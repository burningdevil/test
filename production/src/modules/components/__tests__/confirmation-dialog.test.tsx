import * as React from 'react'
import { mount } from 'enzyme'
import { ConfirmationDialog } from '../common-components/confirmation-dialog/confirmation-dialog'


const mockProps = {
  isConfirmationDialogDisplayed: true,
  triggerAction: jest.fn(),
  closeDialog: jest.fn(),
  wordings: {
    title: 'Delete Confirmation',
    actionButtonText: 'Delete',
    cancelButtonText: 'Cancel',
    summaryText: 'Items to be deleted',
    detailText: 'This operation cannot be undone.'
  }
}

describe('Delete Confirmation Dialog Snapshot Test Set', () => {
  it('Default Confirmation Dialoig Snapshot Test', () => {
    const wrapper = mount(<ConfirmationDialog {...mockProps} />)

    expect(wrapper.debug()).toMatchSnapshot()
  })
})
