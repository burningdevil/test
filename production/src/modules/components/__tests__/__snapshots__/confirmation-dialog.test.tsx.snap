// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Confirmation Dialog Snapshot Test Set Default Confirmation Dialoig Snapshot Test 1`] = `
"<ConfirmationDialog isConfirmationDialogDisplayed={true} triggerAction={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} closeDialog={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} wordings={{...}}>
  <div id={[undefined]} className=\\"confirmation-dialog-wrapper\\">
    <InformationDialog isOpen={true} title=\\"Delete Confirmation\\" hideTitle={false} content={{...}} buttonConfig={{...}} appElement={{...}} className=\\"confirmation-dialog\\" ariaLabel=\\"Delete Confirmation\\" theme=\\"\\" onClose={[Function: handleWindowClose]} ariaLabeledby=\\"\\">
      <div className=\\"information-dialog-panel\\" id=\\"information-dialog-panel confirmation-dialog\\">
        <Dialog isOpen={true} title={{...}} hideTitle={false} content={{...}} footer={{...}} appElement={{...}} className=\\"information-dialog confirmation-dialog\\" theme=\\"\\" ariaLabel=\\"Delete Confirmation\\" ariaLabeledby=\\"\\" hideFooter={false}>
          <Modal isOpen={true} appElement={{...}} className=\\"mstr-rc-dialog information-dialog confirmation-dialog\\" overlayClassName=\\"mstr-rc-dialog-curtain mstr--dialog-theme\\" contentLabel={{...}} shouldCloseOnOverlayClick={false} aria={{...}} portalClassName=\\"ReactModalPortal\\" bodyOpenClassName=\\"ReactModal__Body--open\\" role=\\"dialog\\" ariaHideApp={true} closeTimeoutMS={0} shouldFocusAfterRender={true} shouldCloseOnEsc={true} shouldReturnFocusAfterClose={true} preventScroll={false} parentSelector={[Function: parentSelector]} overlayElement={[Function: overlayElement]} contentElement={[Function: contentElement]}>
            <Portal containerInfo={{...}}>
              <ModalPortal defaultStyles={{...}} isOpen={true} appElement={{...}} className=\\"mstr-rc-dialog information-dialog confirmation-dialog\\" overlayClassName=\\"mstr-rc-dialog-curtain mstr--dialog-theme\\" contentLabel={{...}} shouldCloseOnOverlayClick={false} aria={{...}} portalClassName=\\"ReactModalPortal\\" bodyOpenClassName=\\"ReactModal__Body--open\\" role=\\"dialog\\" ariaHideApp={true} closeTimeoutMS={0} shouldFocusAfterRender={true} shouldCloseOnEsc={true} shouldReturnFocusAfterClose={true} preventScroll={false} parentSelector={[Function: parentSelector]} overlayElement={[Function: overlayElement]} contentElement={[Function: contentElement]} style={{...}}>
                <div className=\\"ReactModal__Overlay mstr-rc-dialog-curtain mstr--dialog-theme\\" style={{...}} onClick={[Function (anonymous)]} onMouseDown={[Function (anonymous)]}>
                  <div id={[undefined]} style={{...}} className=\\"ReactModal__Content mstr-rc-dialog information-dialog confirmation-dialog\\" tabIndex=\\"-1\\" onKeyDown={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onClick={[Function (anonymous)]} role=\\"dialog\\" aria-label=\\"Delete Confirmation\\" aria-modal={true} aria-labelledby=\\"\\" data-testid={[undefined]}>
                    <div className=\\"mstr-rc-dialog-title\\">
                      <div className=\\"information-dialog title\\">
                        <div className=\\"information-dialog title label\\">
                          Delete Confirmation
                        </div>
                        <div className=\\"information-dialog title close-button\\" onClick={[Function: onClick]} role=\\"button\\" tabIndex={0} onKeyDown={[Function: onKeyDown]}>
                           
                          <i className=\\"ws-icon-18 ws-icon-close-button-cross\\" />
                           
                        </div>
                      </div>
                    </div>
                    <div className=\\"mstr-rc-dialog-content\\">
                      <div className=\\"confirmation-dialog-dialog-container\\">
                        <div className=\\"confirmation-dialog-warning-icon\\">
                          <i className=\\"ws-icon-60 ws-icon-mstr-warning\\" />
                        </div>
                        <div className=\\"confirmation-dialog-content-container\\">
                          <div tabIndex={0} className=\\"confirmation-dialog-summary\\">
                            <span>
                              Items to be deleted
                            </span>
                          </div>
                          <div tabIndex={0} className=\\"confirmation-dialog-details\\">
                            <span>
                              This operation cannot be undone.
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div className=\\"mstr-rc-dialog-footer\\">
                      <div className=\\"footer button-container\\">
                        <Button title=\\"Delete\\" className=\\"confirmation-dialog-action-button\\" onClick={[Function: handleAction]} type=\\"primary\\" size=\\"regular\\">
                          <div className=\\"mstr-button-container\\">
                            <Button data-testid=\\"\\" block={[undefined]} className=\\"confirmation-dialog-action-button mstr-button mstr-button__primary-type mstr-button__regular-size\\" id=\\"\\" size=\\"middle\\" type=\\"primary\\" disabled={false} onClick={[Function: handleAction]}>
                              <ForwardRef disabled={false}>
                                <InternalWave disabled={false}>
                                  <button data-testid=\\"\\" id=\\"\\" onClick={[Function: handleClick]} type=\\"button\\" className=\\"ant-btn ant-btn-primary confirmation-dialog-action-button mstr-button mstr-button__primary-type mstr-button__regular-size\\" disabled={false}>
                                    <LoadingIcon existIcon={false} prefixCls=\\"ant-btn\\" loading={false}>
                                      <CSSMotion visible={false} motionName=\\"ant-btn-loading-icon-motion\\" removeOnLeave={true} onAppearStart={[Function: getCollapsedWidth]} onAppearActive={[Function: getRealWidth]} onEnterStart={[Function: getCollapsedWidth]} onEnterActive={[Function: getRealWidth]} onLeaveStart={[Function: getRealWidth]} onLeaveActive={[Function: getCollapsedWidth]}>
                                        <DomWrapper />
                                      </CSSMotion>
                                    </LoadingIcon>
                                    <span>
                                      Delete
                                    </span>
                                  </button>
                                </InternalWave>
                              </ForwardRef>
                            </Button>
                          </div>
                        </Button>
                        <Button title=\\"Cancel\\" className=\\"confirmation-dialog-cancel-button\\" onClick={[Function: handleCancel]}>
                          <div className=\\"mstr-button-container\\">
                            <Button data-testid=\\"\\" block={[undefined]} className=\\"confirmation-dialog-cancel-button mstr-button mstr-button__regular-type mstr-button__regular-size\\" id=\\"\\" size=\\"middle\\" type=\\"default\\" disabled={false} onClick={[Function: handleCancel]}>
                              <ForwardRef disabled={false}>
                                <InternalWave disabled={false}>
                                  <button data-testid=\\"\\" id=\\"\\" onClick={[Function: handleClick]} type=\\"button\\" className=\\"ant-btn ant-btn-default confirmation-dialog-cancel-button mstr-button mstr-button__regular-type mstr-button__regular-size\\" disabled={false}>
                                    <LoadingIcon existIcon={false} prefixCls=\\"ant-btn\\" loading={false}>
                                      <CSSMotion visible={false} motionName=\\"ant-btn-loading-icon-motion\\" removeOnLeave={true} onAppearStart={[Function: getCollapsedWidth]} onAppearActive={[Function: getRealWidth]} onEnterStart={[Function: getCollapsedWidth]} onEnterActive={[Function: getRealWidth]} onLeaveStart={[Function: getRealWidth]} onLeaveActive={[Function: getCollapsedWidth]}>
                                        <DomWrapper />
                                      </CSSMotion>
                                    </LoadingIcon>
                                    <span>
                                      Cancel
                                    </span>
                                  </button>
                                </InternalWave>
                              </ForwardRef>
                            </Button>
                          </div>
                        </Button>
                      </div>
                    </div>
                  </div>
                </div>
              </ModalPortal>
            </Portal>
          </Modal>
        </Dialog>
      </div>
    </InformationDialog>
  </div>
</ConfirmationDialog>"
`;
